fix(web): add early version check in Language.load()

wtdlee opened the pull request:
## Summary

Add ABI version compatibility check immediately after loading a WASM language file in `Language.load()`.

## Problem

Currently, when loading a WASM file built with an incompatible `tree-sitter-cli` version:

1. `Language.load()` succeeds
2. `parser.setLanguage()` fails with: `Incompatible language version X. Compatibility range Y through Z.`

This is confusing because:
- The error occurs at `setLanguage()`, not at load time
- The error message doesn't indicate the root cause (incompatible WASM build)
- Users don't know how to fix it

This commonly affects users of third-party WASM packages like `tree-sitter-wasms` that may be built with older `tree-sitter-cli` versions.

## Solution

Add version check in `Language.load()` right after loading the WASM module:

```typescript
const version = language.abiVersion;
if (version < MIN_COMPATIBLE_VERSION || LANGUAGE_VERSION < version) {
  throw new Error(
    `Incompatible language version ${version}. ` +
    `Compatibility range ${MIN_COMPATIBLE_VERSION} through ${LANGUAGE_VERSION}. ` +
    `The WASM file may have been built with an incompatible version of tree-sitter-cli. ` +
    `Please regenerate the WASM file using a compatible tree-sitter-cli version.`
  );
}
```

## Benefits

1. **Earlier failure** - Fails at load time instead of `setLanguage()`
2. **Better error message** - Clearly indicates the WASM file might be built with incompatible CLI
3. **Actionable guidance** - Tells users to regenerate the WASM file

## Changes

- `lib/binding_web/src/language.ts`:
  - Import `LANGUAGE_VERSION` and `MIN_COMPATIBLE_VERSION` from `parser.ts`
  - Add version compatibility check after loading WASM
  - Add `@throws` JSDoc annotation

## Related

- #5171 - Issue describing the compatibility problem
- #5172 - Documentation PR for version compatibility

ObserverOfTime commented on lib/binding_web/src/language.ts:
```diff
@@ -263,6 +265,19 @@ export class Language {
+    // Early version compatibility check
+    const version = language.abiVersion;
+    if (version < MIN_COMPATIBLE_VERSION || LANGUAGE_VERSION < version) {
+      throw new Error(
```
```suggestion
      throw new RangeError(
```

ObserverOfTime commented on lib/binding_web/src/language.ts:
```diff
@@ -263,6 +265,19 @@ export class Language {
+        `Incompatible language version ${version}. ` +
+        `Compatibility range ${MIN_COMPATIBLE_VERSION} through ${LANGUAGE_VERSION}. ` +
+        `The WASM file may have been built with an incompatible version of tree-sitter-cli. ` +
+        `Please regenerate the WASM file using a compatible tree-sitter-cli version.`
```
```suggestion
        `The Wasm file may have been built with an incompatible version of tree-sitter-cli. `
```

wtdlee commented:
Thank you for the review @ObserverOfTime! 

Applied all suggestions

- Changed `Error` to `RangeError`
- Fixed `WASM` → `Wasm` casing
- Removed the regeneration hint for a cleaner message

Let me know if there's anything else!

WillLillis commented:
Thanks for investigating this. The follow-up analysis in #5171 established that the affected Wasm files fail inside Emscripten’s loader because they use the legacy dylink section format, before a `Language` is constructed or its ABI can be checked. Genuine ABI mismatches are already checked by `Parser.setLanguage`, so duplicating that validation in `Language.load` would not address the reported failure.
